### ISSUE TITLE
Info file has tons of extensions

### ISSUE THREAD

**Issue by cathoderaydude** [libretro/scummvm#71]
I hardly even want to bug you with an issue for this, since I think I know the answer already, but I couldn't find anyone else asking about it and maybe it'll be helpful to somebody in the future.

The info file for the scummvm core has an enormous number of filetypes in the supported_extensions directive, including things like .C, .IMG, .TXT, .MP3, .CFG, and .TTF. The effect is that if scummvm is installed, then an awful lot of odd files will pass the "Filter unknown extensions" check and show up in Load Content, cluttering things up if your disk is a mess. **Am I right in assuming this is necessary and unavoidable?**

Obviously it's not a cause celebre; virtually everyone (except me) is using well-maintained ROM folders and will never even know this is happening. And I assume it's necessary because ScummVM has to access the raw files of many different games straight from the disk, and if it doesn't specify all this stuff in the info file, then the file access APIs will refuse to expose any of those files to the core. And this is probably unfixable without extending the APIs with a "no_seriously_show_me_everything" flag, which isn't worth the impact since virtually nothing else will ever need this functionality.

If that's all correct then go ahead and close, ty! 👍

**Comment by spleen1981** [comment]
Hi, all the extensions shown in the core info file are absolutely not needed for the core itself, it would work perfectly even with no extension indicated at all (though I would indicate just the `.scummvm`).
Those extensions are only needed to let the external `ScummVM.dat`  from [libretro-database](https://github.com/libretro/libretro-database) work to generate a ScummVM retroarch playlist from the frontend.

Actually this is not the right way to generate a ScummVM retroarch playlist (info about how to do this properly are shown in the help section of the internal ScummVM Launcher, soon or later I will update libretro docs as well) and the `ScummVM.dat` is not handled by me or the ScummVM team.
Anyways I don't like to break this legacy feature unilaterally core-side, I will do if `ScummVM.dat` is dropped from libretro-database or if it becomes evidently obsolete or abandoned.
I will leave the issue opened as an enhancement, pending on `ScummVM.dat` removal, to be re-evaluated in the future.

To solve your issue locally, you may open your `scummvm_libretro.info` in the core info folder and remove all unneeded extensions, though the file would be overwritten during next update.

**Comment by spleen1981** [comment]
> Those extensions are only needed to let the external ScummVM.dat from [libretro-database](https://github.com/libretro/libretro-database) work to generate a ScummVM retroarch playlist from the frontend.

Actually the retroarch scanner will work without those extensions  indicated in the info file as well.
Dropping those will only affect the manual "load content" feature from retroarch, as only `.scummvm` files will be now loaded automatically by scummvm core on manual content load from frontend. Previously an arbitrary subset of extensions of all possible gamefiles was allowed to be loaded directly in scummvm, including e.g. `.avi` files, loaded in scummvm if ffmpeg was disabled, and all sorts of exotic extensions...
Extensions other than `.scummvm` can still be loaded by scummvm via playlist, hence playlists created with Retroarch Scanner will still work (though this is **NOT** the recommended way to create scummvm playlists).
Hence support for extensions other than .scummvm in the core info file is now removed 92597f299616f20a25fb2d208f096f927516af8e
